lib/recursive_match.ex

```diff
@@ -147,7 +147,7 @@ defmodule RecursiveMatch do
       right: %{a: 1}
   """
 
-  @spec assert_match(term, term, list | nil) :: boolean
+  @spec assert_match(term, term, list | nil) :: binary
```

Comment:
The @spec for the function does not match the success typing of the function.

Function:
RecursiveMatch.MACRO-assert_match/4

Success typing:
@spec MACRO-assert_match(_, _, _, nil | Keyword.t() | map()) ::
  {:__block__, [], [{:= | {_, _, _}, [], [any(), ...]}, ...]}